source/simulator/core.cc

```diff
@@ -1694,6 +1700,8 @@ namespace aspect
     // we disable automatic summary printing so that it won't happen when
     // throwing an exception. Therefore, we have to do this manually here:
     computing_timer.print_summary ();
+
+    CitationInfo::print_info_block (pcout);
```

Comment:
I wonder how often users (i) actually run the program to termination, and (ii) then look at the very bottom of their log file. But I can't come up with a better point to print stuff either, other than right at the beginning.

Reply:
At the moment, we have the citation information right at the beginning, so we want to make sure that we don't output contradicting information. But we could output it twice, at the beginning and the end, or say in of of these places "the citation information can be found at the beginning/end of the log file".

Reply:
We could also output it at the beginning of `run()`. I think that by that point, we have all of the information.

Reply:
it is already printed at start and end.